base/strings/util.jl

```diff
@@ -513,6 +513,138 @@ function rpad(
     r == 0 ? stringfn(s, p^q) : stringfn(s, p^q, first(p, r))
 end
 
+"""
+    rtrunc(str::AbstractString, maxwidth::Int, replace_str::AbstractString = "…")
+
+Truncate `str` to at most `maxwidth` columns (as estimated by [`textwidth`](@ref)), replacing the last characters
+with `replacement` if necessary. The default replacement string is "…".
+
+# Examples
+```jldoctest
+julia> s = rtrunc("🍕🍕 I love 🍕", 10)
+"🍕🍕 I lo…"
+
+julia> textwidth(s)
+10
+
+julia> rtrunc("foo", 3)
+"foo"
+```
+
+!!! compat "Julia 1.12"
+    This function was added in Julia 1.12.
+
+See also [`ltrunc`](@ref) and [`ctrunc`](@ref).
+"""
+function rtrunc(str::AbstractString, maxwidth::Int, replacement::Union{AbstractString,Char} = '…')
+    ret = string_truncate_boundaries(str, maxwidth, replacement, Val(:right))
+    if isnothing(ret)
+        return str
+    else
+        left, _ = ret::Tuple{Int,Int}
+        @views return str[begin:left] * replacement
+    end
+end
+
+"""
+    ltrunc(str::AbstractString, maxwidth::Int, replace_str::AbstractString = "…")
+
+Truncate `str` to at most `maxwidth` columns (as estimated by [`textwidth`](@ref)), replacing the first characters
+with `replacement` if necessary. The default replacement string is "…".
+
+# Examples
+```jldoctest
+julia> s = ltrunc("🍕🍕 I love 🍕", 10)
+"…I love 🍕"
+
+julia> textwidth(s)
+10
+
+julia> ltrunc("foo", 3)
+"foo"
+```
+
+!!! compat "Julia 1.12"
+    This function was added in Julia 1.12.
+
+See also [`rtrunc`](@ref) and [`ctrunc`](@ref).
+"""
+function ltrunc(str::AbstractString, maxwidth::Int, replacement::Union{AbstractString,Char} = '…')
```

Comment:
```suggestion
function ltrunc(str::AbstractString, maxwidth::Integer, replacement::Union{AbstractString,AbstractChar} = '…')
```
and pass `Int(maxwidth)` to `string_truncate_boundaries`.

Similarly below.